code/game/objects/items/shuttle_creator.dm

```diff
@@ -0,0 +1,365 @@
+	if(!(get_offset_target_turf(targetTurf, 0, 1) in loggedTurfs))
+		if(position != null)
+			return null
+		position = NORTH
+	if(!(get_offset_target_turf(targetTurf, 0, -1) in loggedTurfs))
+		if(position != null)
+			return null
+		position = SOUTH
+	if(!(get_offset_target_turf(targetTurf, 1, 0) in loggedTurfs))
+		if(position != null)
+			return null
+		position = EAST
+	if(!(get_offset_target_turf(targetTurf, -1, 0) in loggedTurfs))
+		if(position != null)
+			return null
+		position = WEST
+	return position
+
+/obj/item/shuttle_creator/proc/invertDir(input_dir)
+	if(input_dir == NORTH)
+		return SOUTH
+	else if(input_dir == SOUTH)
+		return NORTH
+	else if(input_dir == EAST)
+		return WEST
+	else if(input_dir == WEST)
+		return EAST
+	return null
+
+/obj/item/shuttle_creator/proc/shuttle_create_docking_port(atom/target, mob/user)
+
+	if(loggedTurfs.len == 0 || !recorded_shuttle_area)
+		to_chat(user, "<span class='warning'>Invalid shuttle, restarting bluespace systems...</span>")
+		return FALSE
+
+	var/datum/map_template/shuttle/new_shuttle = new /datum/map_template/shuttle()
+
+	var/obj/docking_port/mobile/port = new /obj/docking_port/mobile(get_turf(target))
+	var/obj/docking_port/stationary/stationary_port = new /obj/docking_port/stationary(get_turf(target))
+	port.callTime = 100
+	port.dir = 1	//Point away from space.
+	port.shuttle_id = "custom_[GLOB.custom_shuttle_count]"
+	linkedShuttleId = port.shuttle_id
+	port.ignitionTime = 25
+	port.name = "Custom Shuttle"
+	port.port_direction = 2
+	port.preferred_direction = 4
+	port.area_type = recorded_shuttle_area
+	port.can_move_docking_ports = TRUE
+
+	stationary_port.area_type = overwritten_area
+
+	var/portDirection = getNonShuttleDirection(get_turf(port))
+	var/invertedDir = invertDir(portDirection)
+	if(!portDirection || !invertedDir)
+		to_chat(usr, "<span class='warning'>Shuttle creation aborted, docking airlock must be on an external wall. Please select a new airlock.</span>")
+		port.Destroy()
+		stationary_port.Destroy()
+		linkedShuttleId = null
+		return FALSE
+	port.dir = invertedDir
+	port.port_direction = portDirection
+
+	if(!calculate_bounds(port))
+		to_chat(usr, "<span class='warning'>Bluespace calculations failed, please select a new airlock.</span>")
+		port.Destroy()
+		stationary_port.Destroy()
+		linkedShuttleId = null
+		return FALSE
+
+	port.shuttle_areas = list()
+	//var/list/all_turfs = port.return_ordered_turfs(port.x, port.y, port.z, port.dir)
+	var/list/all_turfs = loggedTurfs
+	for(var/i in 1 to all_turfs.len)
+		var/turf/curT = all_turfs[i]
+		var/area/cur_area = curT.loc
+		//Add the area to the shuttle <3
+		if(istype(cur_area, recorded_shuttle_area))
+			if(istype(curT, /turf/open/space))
+				continue
+			if(length(curT.baseturfs) < 2)
+				continue
+			//Add the shuttle base shit to the shuttle
+			curT.baseturfs.Insert(3, /turf/baseturf_skipover/shuttle)
+			port.shuttle_areas[cur_area] = TRUE
+
+	port.linkup(new_shuttle, stationary_port)
```

Comment:
what is going on with this proc? linkup only has one arg it doesn't care about the map template at all. also why do we have an empty map template here